python/ray/serve/scripts.py

```diff
@@ -378,3 +381,51 @@ def delete(address: str, yes: bool):
     cli_logger.newline()
     cli_logger.success("\nSent delete request successfully!\n")
     cli_logger.newline()
+
+
+@cli.command(
+    short_help="Writes a Pipeline's config file.",
+    help=(
+        "Imports the DeploymentNode or DeploymentFunctionNode at IMPORT_PATH "
+        "and generates a structured config for it that can be used by "
+        "`serve deploy` or the REST API. "
+    ),
+)
+@click.option(
+    "--app-dir",
+    "-d",
+    default=".",
+    type=str,
+    help=APP_DIR_HELP_STR,
+)
+@click.option(
+    "--output-path",
+    "-o",
+    default=None,
+    type=str,
+    help=(
+        "Local path where the output config will be written in YAML format. "
+        "If not provided, the config will be printed to STDOUT."
+    ),
+)
+@click.argument("import_path")
+def build(app_dir: str, output_path: Optional[str], import_path: str):
+    sys.path.insert(0, app_dir)
+
+    node: Union[DeploymentNode, DeploymentFunctionNode] = import_attr(import_path)
+    if not isinstance(node, (DeploymentNode, DeploymentFunctionNode)):
+        raise TypeError(
+            f"Expected '{import_path}' to be DeploymentNode or "
+            f"DeploymentFunctionNode, but got {type(node)}."
+        )
+
+    app = serve.build(node)
+
+    if output_path is not None:
+        if not output_path.endswith(".yaml"):
+            raise ValueError("FILE_PATH must end with '.yaml'.")
+
+        with open(output_path, "w") as f:
+            app.to_yaml(f)
+    else:
+        print(app.to_yaml(), end="")
```

Comment:
not for this PR, but as a follow up it will be much more actionable to print other instructions for next step, such as providing a file path, or calling other clis to deployment the yaml file.